src/manager/manager.py

```diff
@@ -62,10 +38,47 @@ def _update_manager_vehicle_list(manager: Manager, vehicles: list[Vehicle]):
         # append if not in list and inside radius
         if not vehicle_in_list and distance_to_vehicle <= manager.radius: 
             manager.vehicles.append(vehicle)
+            new_vehicle = True
 
         # remove if in list and outside radius
         elif vehicle_in_list and distance_to_vehicle > manager.radius: 
             manager.vehicles.remove(vehicle)
+    return new_vehicle
+
+def get_collisions(manager: Manager, cur_time: float):
+    collisions = []
+    vehicle_pairs = combinations(manager.vehicles, 2)
+    threshold_distance = 2.5 # will be replaced by Alex's defined safety radius
```

Comment:
threshold_distance=2.5 doesn't seem to reflect the current simulation visual. Consider threshold_distance=4, the one we tested.